hack/verify-prometheus-imports.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script validates that only a restricted set of packages are importing
+# github.com/prometheus/*
+
+# NOTE: this is not the same as verify-imports which can only verify
+# that within a particular package the imports made are allowed.
+#
+# This is also not the same thing as verify-import-boss, which is pretty
+# powerful for specifying restricted imports but does not scale to checking
+# the entire source tree well and is only enabled for specific packages.
+#
+# See: https://github.com/kubernetes/kubernetes/issues/99876
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+source "${KUBE_ROOT}/hack/lib/util.sh"
+
+# See: https://github.com/kubernetes/kubernetes/issues/89267
+allowed_prometheus_importers=(
+  ./cluster/images/etcd-version-monitor
+  ./pkg/volume/util/operationexecutor
+  ./staging/src/k8s.io/component-base/metrics
+  ./staging/src/k8s.io/component-base/metrics/legacyregistry
+  ./staging/src/k8s.io/component-base/metrics/testutil
+  ./test/e2e/apimachinery
+  ./test/e2e_node
+  ./test/integration/apiserver/flowcontrol
+  ./test/integration/metrics
+)
```

Comment:
This is the current list of importers. 
It also matches the previous visibility rule, though is more specific for component-base/metrics.

https://github.com/kubernetes/kubernetes/blob/a1f2787897a81c375a4de28482b5eb649851800a/build/visible_to/BUILD#L431-L444